casper/src/main/scala/coop/rchain/casper/engine/LfsBlockRequester.scala

```diff
@@ -82,7 +87,8 @@ object LfsBlockRequester {
         // Calculate new minimum height if latest message
         //  - we need parents of latest message so it's `-1`
         val newLowerBound = if (isLatest) Math.min(height - 1, lowerBound) else lowerBound
-        val newSt         = d + ((k, Received))
+        val newSt =
+          latestReplacement.foldLeft(d + ((k, Received)))((acc, k) => acc + (k -> Requested))
```

Comment:
1. Why is here `latestReplacement` added as received?
2. Where is `latestReplacement` block height captured for min height calculation?

Reply:
here lower bound overwritten https://github.com/rchain/rchain/blob/cafbda84eafb97153292f063fe3b7114adc0e3f0/casper/src/main/scala/coop/rchain/casper/engine/LfsBlockRequester.scala#L84
>>
and here block is rejected if it is lower, right?
https://github.com/rchain/rchain/blob/cafbda84eafb97153292f063fe3b7114adc0e3f0/casper/src/main/scala/coop/rchain/casper/engine/LfsBlockRequester.scala#L208-L223

Reply:
Oh, I see. It will be calculated in the same way when this _latest replacement_ is received.

Maybe we can simplify this even more. If replacement message is added to `latest` on the begging, rest of the function can stay unchanged.

```scala
if (isReq) {
  // Replace latest message with replacement if supplied
  val latestWithR = latestReplacement.map(latest - k + _).getOrElse(latest)
  // Remove message from the set of latest messages (if exists)
  val newLatest    = latestWithR - k
  val isLatest     = latest != newLatest
```
In that way we don't need to take care of Request created for new latest replacement and leave existing mechanism to work the same. Old latest will not even be detected as latest because it's replaced by new latest.

Reply:
I pushed this change and removed unnecessary calculation in initializing engine.